### PR TITLE
Ipavlopoulos  tags parse

### PR DESCRIPTION
**Pull request by ipavlopoulos**
Also parsing tags and saving in JSON for IU_XRAY

### CODE REVIEW

File: SiVL19/get_iu_xray.py [hunk 1]
```diff
@@ -49,47 +52,58 @@ def split_cases(reports_images, reports_text, keys, filename):
...
+            if tags is not None:
+                major_tags = [t.text for t in tags.findall("major")]
+                auto_tags = [t.text for t in tags.findall("automatic")]
+
+            for image in images:
+                iid = image.get("id") + ".png"
+                images_captions[iid] = caption
+                img_ids.append(iid)
+                images_major_tags[iid] = major_tags
+                images_auto_tags[iid] = auto_tags
```
Comment (vasilikikou) [inline SiVL19/get_iu_xray.py]: Some reports do not contain automatic tags. Do we want to save the images of these reports?

File: SiVL19/get_peir_gross.py [hunk 2]
```diff
@@ -86,9 +87,12 @@ def split_images(images, keys, filename):
 				filename = image.get("alt")
 				image_src = image.get("src")
 				description = image.get("title").replace("\r\n", " ")
+				image_captions[filename] = description
+
+				tags_container = image_soup.find("div", {"id":"Tags"})
+				tags = [tag.string for tag in tags_container.findChildren("a")]
```
Comment (vasilikikou) [inline SiVL19/get_peir_gross.py]: Tag "gross" is present in all the images. Maybe we should not include it?

File: SiVL19/get_iu_xray.py [hunk 3]
```diff
@@ -117,4 +139,4 @@ def split_cases(reports_images, reports_text, keys, filename):
 test_path = "iu_xray/test_images.tsv"
 
 split_cases(reports_with_images, text_of_reports, train_keys, train_path)
-split_cases(reports_with_images, text_of_reports, test_keys, test_path)
\ No newline at end of file
+split_cases(reports_with_images, text_of_reports, test_keys, test_path)
```
Comment (vasilikikou) [inline SiVL19/get_iu_xray.py]: Should we also add split for the tags?

File: SiVL19/get_iu_xray.py (hunk 1 above)
Comment (ipavlopoulos) [reply SiVL19/get_iu_xray.py]: We want all images associated with a caption. Why should the caption existence matter?

File: SiVL19/get_peir_gross.py (hunk 2 above)
Comment (ipavlopoulos) [reply SiVL19/get_peir_gross.py]: Sounds like a pre-processing step.

File: SiVL19/get_iu_xray.py (hunk 3 above)
Comment (ipavlopoulos) [reply SiVL19/get_iu_xray.py]: I am not sure I am following your thought. How would that work exactly?

Comment (vasilikikou) [reply SiVL19/get_iu_xray.py]: The same way it works for the captions. We could create a dictionary tags_of_reports with the report as key and the tags as value and then call split_cases.

Comment (ipavlopoulos) [reply SiVL19/get_iu_xray.py]: I am still failing to see the need/benefits of this step.

**Pull request merged**